sandbox_dns_unix.go

```diff
@@ -275,7 +275,14 @@ func (sb *sandbox) updateDNS(ipv6Enabled bool) error {
 	if err != nil {
 		return err
 	}
-	if err = ioutil.WriteFile(tmpHashFile.Name(), []byte(newRC.Hash), filePerm); err != nil {
```

Comment:
In this case we should not `defer` since the file will get renamed at the end. Before the file is renamed the file should be synced and closed.